### PR TITLE
feat: add integers to io-ts-numbers

### PR DESCRIPTION
**Pull request by paddyquinn**
Add corresponding tests/docs as well.

### CODE REVIEW

File: packages/fast-check-numbers/src/ZeroInt.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import * as fc from 'fast-check'
+import * as t from 'io-ts'
+import { Zero } from 'io-ts-numbers'
+
+/**
+ * @since 1.1.0
+ */
+export const ZeroIntArbitrary: fc.Arbitrary<Zero & t.Int> = fc
+  .oneof(fc.constant(0), fc.constant(-0))
+  .filter((n): n is Zero & t.Int => n === 0 && Number.isInteger(n))
```
Comment (EricCrosson) [inline packages/fast-check-numbers/src/ZeroInt.ts:14]:
```suggestion
  .filter(ZeroInt.is)
```

Here I would prefer using the `ZeroInt` codec as a type-guard, to provide the guarantee that any value provided by the arbitrary will be decodable by the `ZeroInt` codec. This way, even if we get the logic in line 13 wrong, worst-case is we filter out the incorrect values and perhaps return a subset of the true `ZeroInt` value space.

Wait a minute -- zero is by definition an integer so isn't this entire codec moot? I think we can remove this entire file and just offer `Zero` via the package API. We can also remove the ZeroInt codec in io-ts-numbers

File: packages/fast-check-numbers/test/property/test-zero-int.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { testProp } from 'ava-fast-check'
```
Comment (EricCrosson) [inline packages/fast-check-numbers/test/property/test-zero-int.ts:1]: Can remove this file by the same logic as above

File: packages/io-ts-numbers/src/ZeroInt.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/**
```
Comment (EricCrosson) [inline packages/io-ts-numbers/src/ZeroInt.ts:1]: Can remove this file by the same logic as above

File: packages/io-ts-numbers/src/ZeroInt.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+import * as t from 'io-ts'
+
+import { Int } from './Int'
+import { Zero } from './Zero'
+
+/**
+ * @since 0.1.0
+ */
+export const ZeroInt = t.intersection([Zero, Int], 'ZeroInt')
```
Comment (EricCrosson) [inline packages/io-ts-numbers/src/ZeroInt.ts:13]: It strikes me that we can encode the knowledge the `zero` is always an integer by moving this intersection to the existing `Zero.ts` file

File: packages/io-ts-numbers/src/NonPositiveIntFromString.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+ */
+
+import { IntFromString } from 'io-ts-types'
+
+import { NonPositiveInt } from './NonPositiveInt'
+
+/**
+ * @since 0.1.0
+ */
+export const NonPositiveIntFromString = IntFromString.pipe(NonPositiveInt)
```
Comment (EricCrosson) [inline packages/io-ts-numbers/src/NonPositiveIntFromString.ts:12]:
```suggestion
export const NonPositiveIntFromString = NumberFromString.pipe(NonPositiveInt)
```

File: packages/io-ts-numbers/src/NonZeroIntFromString.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+ */
+
+import { IntFromString } from 'io-ts-types'
+
+import { NonZeroInt } from './NonZeroInt'
+
+/**
+ * @since 0.1.0
+ */
+export const NonZeroIntFromString = IntFromString.pipe(NonZeroInt)
```
Comment (EricCrosson) [inline packages/io-ts-numbers/src/NonZeroIntFromString.ts:12]:
```suggestion
export const NonZeroIntFromString = NumberFromString.pipe(NonZeroInt)
```

File: packages/io-ts-numbers/src/PositiveIntFromString.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+ */
+
+import { IntFromString } from 'io-ts-types'
+
+import { PositiveInt } from './PositiveInt'
+
+/**
+ * @since 0.1.0
+ */
+export const PositiveIntFromString = IntFromString.pipe(PositiveInt)
```
Comment (EricCrosson) [inline packages/io-ts-numbers/src/PositiveIntFromString.ts:12]:
```suggestion
export const PositiveIntFromString = NumberFromString.pipe(PositiveInt)
```

File: packages/io-ts-numbers/src/ZeroIntFromString.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+ */
+
+import { IntFromString } from 'io-ts-types'
+
+import { ZeroInt } from './ZeroInt'
+
+/**
+ * @since 0.1.0
+ */
+export const ZeroIntFromString = IntFromString.pipe(ZeroInt)
```
Comment (EricCrosson) [inline packages/io-ts-numbers/src/ZeroIntFromString.ts:12]:
```suggestion
export const ZeroIntFromString = NumberFromString.pipe(ZeroInt)
```

**Review by EricCrosson — CHANGES_REQUESTED** [review]

File: packages/io-ts-numbers/src/Int.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+ * @since 0.1.0
+ */
+
+import * as t from 'io-ts'
+import { withEncode } from 'io-ts-types'
+
+/**
+ * @since 0.1.0
+ */
+export const Int = withEncode(t.Int, (_) => _)
```
[the remark this answers is not in the document]
Comment (paddyquinn) [reply packages/io-ts-numbers/src/Int.ts:11]: Moving to `NumberFromString` instead of `IntFromString` renders this codec moot anyway so I just removed it.

**Review by EricCrosson — APPROVED** [review]
Looks great to me. Thanks for adding these missing codecs, @paddyquinn

**Pull request merged**